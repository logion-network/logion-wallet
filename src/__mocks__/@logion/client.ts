import { CheckResultType } from "@logion/client/dist/CollectionItem.js";
import { DefaultSignAndSendStrategy } from "@logion/client/dist/Signer.js";

import {
    LogionClient,
    AccountTokens,
    toIsoString,
    fromIsoString,
    ClosedCollectionLoc,
    DraftRequest,
    EditableRequest,
    OpenLoc,
    LocRequestState,
    LocsState,
    HashOrContent,
} from '../LogionClientMock';

export {
    LogionClient,
    AccountTokens,
    toIsoString,
    fromIsoString,
    CheckResultType,
    ClosedCollectionLoc,
    DraftRequest,
    EditableRequest,
    OpenLoc,
    DefaultSignAndSendStrategy,
    LocRequestState,
    LocsState,
    HashOrContent,
}

export { MimeType } from "@logion/client/dist/Mime.js";
