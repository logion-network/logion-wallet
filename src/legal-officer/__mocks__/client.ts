import {
    closeLoc,
    deleteLink,
    voidLoc,
    getVerifiedThirdPartySelections,
} from "./ClientMock";

export {
    closeLoc,
    deleteLink,
    voidLoc,
    getVerifiedThirdPartySelections,
}
