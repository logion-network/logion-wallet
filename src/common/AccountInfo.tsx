import { UserIdentity as IdentityType, PostalAddress as PostalAddressType } from "@logion/client";

import Identity from "./Identity";
import PostalAddress from "./PostalAddress";
import './AccountInfo.css';
import { BackgroundAndForegroundColors } from "./ColorTheme";

export interface Props {
    label: string
    identity?: IdentityType
    postalAddress?: PostalAddressType
    otherIdentity?: IdentityType
    otherPostalAddress?: PostalAddressType
    colors: BackgroundAndForegroundColors
    squeeze: boolean
    noComparison: boolean
}

export default function AccountInfo(props: Props) {
    return (
        <div className="AccountInfo">
            <Identity
                identity={ props.identity }
                otherIdentity={ props.otherIdentity }
                colors={ props.colors }
                squeeze={ props.squeeze }
                noComparison={ props.noComparison }
            />
            <PostalAddress
                postalAddress={ props.postalAddress }
                otherPostalAddress={ props.otherPostalAddress }
                colors={ props.colors }
                squeeze={ props.squeeze }
                noComparison={ props.noComparison }
            />
        </div>
    )
}
