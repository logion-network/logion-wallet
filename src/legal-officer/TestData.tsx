import { DateTime } from 'luxon';
import { toIsoString } from "@logion/client";
import { ProtectionRequest } from '@logion/client/dist/RecoveryClient.js';

import { DEFAULT_LEGAL_OFFICER, ANOTHER_LEGAL_OFFICER } from "../common/TestData";

export const PENDING_PROTECTION_REQUESTS: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER.address,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER.address,
        decision: {
            rejectReason: null,
            decisionOn: null,
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(DateTime.fromISO('2021-06-10T11:40:00.000', {zone: "utc"})),
        isRecovery: false,
        addressToRecover: null,
        status: "PENDING"
    }
];

export const PROTECTION_REQUESTS_HISTORY: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER.address,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER.address,
        decision: {
                rejectReason: null,
                decisionOn: toIsoString(DateTime.fromISO('2021-06-10T11:40:00.000', {zone: "utc"})),
                locId: "locId"
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(DateTime.fromISO('2021-06-10T11:40:00.000', {zone: "utc"})),
        isRecovery: false,
        addressToRecover: null,
        status: "ACTIVATED"
    }
];

export const RECOVERY_REQUESTS_HISTORY: ProtectionRequest[] = [
    {
        id: "1",
        requesterAddress: "5Ew3MyB15VprZrjQVkpQFj8okmc9xLDSEdNhqMMS5cXsqxoW",
        legalOfficerAddress: DEFAULT_LEGAL_OFFICER.address,
        otherLegalOfficerAddress: ANOTHER_LEGAL_OFFICER.address,
        decision: {
            rejectReason: null,
            decisionOn: toIsoString(DateTime.fromISO('2021-06-10T11:40:00.000', {zone: "utc"})),
            locId: "locId"
        },
        userIdentity: {
            firstName: "John",
            lastName: "Doe",
            email: "john.doe@logion.network",
            phoneNumber: "+1234",
        },
        userPostalAddress: {
            line1: "Place de le République Française, 10",
            line2: "boite 15",
            postalCode: "4000",
            city: "Liège",
            country: "Belgium",
        },
        createdOn: toIsoString(DateTime.fromISO('2021-06-10T11:40:00.000', {zone: "utc"})),
        isRecovery: true,
        addressToRecover: "an-address",
        status: "PENDING"
    }
];
