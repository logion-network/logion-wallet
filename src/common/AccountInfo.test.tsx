import { UserIdentity as IdentityType, PostalAddress as PostalAddressType } from "@logion/client";

import { render } from "../tests";
import { DEFAULT_IDENTITY, COLOR_THEME, DEFAULT_ADDRESS } from "./TestData";
import AccountInfo from "./AccountInfo";

const DIFFERENT_IDENTITY: IdentityType = {
    firstName: "John2",
    lastName: "Doe2",
    email: "john.doe@logion.network",
    phoneNumber: "+1234",
};

const DIFFERENT_ADDRESS: PostalAddressType = {
    line1: "Place de le République Française, 258741257",
    line2: "",
    postalCode: "4000",
    city: "Liège",
    country: "Belgium",
};

test("renders without comparison", () => {
    const tree = render(
        <AccountInfo
            label="some label"
            identity={ DEFAULT_IDENTITY }
            postalAddress={ DEFAULT_ADDRESS }
            colors={ COLOR_THEME.dashboard }
            squeeze={ true }
            noComparison={ true }
        />
    );
    expect(tree).toMatchSnapshot();
});

test("renders and compares with same data", () => {
    const tree = render(
        <AccountInfo
            label="some label"
            identity={ DEFAULT_IDENTITY }
            otherIdentity={ DEFAULT_IDENTITY }
            postalAddress={ DEFAULT_ADDRESS }
            otherPostalAddress={ DEFAULT_ADDRESS }
            colors={ COLOR_THEME.dashboard }
            squeeze={ true }
            noComparison={ false }
        />
    );
    expect(tree).toMatchSnapshot();
});

test("renders and compares with different data", () => {
    const tree = render(
        <AccountInfo
            label="some label"
            identity={ DEFAULT_IDENTITY }
            otherIdentity={ DIFFERENT_IDENTITY }
            postalAddress={ DEFAULT_ADDRESS }
            otherPostalAddress={ DIFFERENT_ADDRESS }
            colors={ COLOR_THEME.dashboard }
            squeeze={ true }
            noComparison={ false }
        />
    );
    expect(tree).toMatchSnapshot();
});
