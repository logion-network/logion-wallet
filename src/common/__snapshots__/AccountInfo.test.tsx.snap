// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and compares with different data 1`] = `
<div
  className="AccountInfo"
>
  <div
    className="ComparableField"
  >
    <div
      className="FormGroup"
      style={
        Object {
          "backgroundColor": "#eff3fe",
        }
      }
    >
      <style>
        
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
      </style>
      <div
        data-testid="accountAddressGroup"
      >
        <label
          className="form-label"
          htmlFor="accountAddress"
        >
          some label
        </label>
        <input
          className="form-control"
          data-testid="accountAddress"
          id="accountAddress"
          name="accountAddress"
          onBlur={[Function]}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value="vQxHAE33LeJYV69GCB4o4YcCgnDu8y99u5hy2751fRdxjX9kz"
        />
      </div>
    </div>
  </div>
  <div
    className="Identity"
  >
    <div
      className="row"
    >
      <div
        className="IdentityCol col-md-6"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="firstNameGroup"
            >
              <label
                className="form-label"
                htmlFor="firstName"
              >
                First Name
              </label>
              <input
                className="form-control is-invalid"
                data-testid="firstName"
                id="firstName"
                name="firstName"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="John"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="IdentityCol col-md-6"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="lastNameGroup"
            >
              <label
                className="form-label"
                htmlFor="lastName"
              >
                Last Name
              </label>
              <input
                className="form-control is-invalid"
                data-testid="lastName"
                id="lastName"
                name="lastName"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="Doe"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="IdentityCol col-md-6"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="emailGroup"
            >
              <label
                className="form-label"
                htmlFor="email"
              >
                Email
              </label>
              <input
                className="form-control"
                data-testid="email"
                id="email"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="john.doe@logion.network"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="IdentityCol col-md-6"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="phoneNumberGroup"
            >
              <label
                className="form-label"
                htmlFor="phoneNumber"
              >
                Phone Number
              </label>
              <input
                className="form-control"
                data-testid="phoneNumber"
                id="phoneNumber"
                name="phoneNumber"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="+1234"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="PostalAddress"
  >
    <h3>
      Address
    </h3>
    <div
      className="row"
    >
      <div
        className="PostalAddressCol col-md-12"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="line1Group"
            >
              <label
                className="form-label"
                htmlFor="line1"
              >
                Line1
              </label>
              <input
                className="form-control is-invalid"
                data-testid="line1"
                id="line1"
                name="line1"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="Place de le République Française, 10"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="PostalAddressCol col-md-12"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="line2Group"
            >
              <label
                className="form-label"
                htmlFor="line2"
              >
                Line2
              </label>
              <input
                className="form-control is-invalid"
                data-testid="line2"
                id="line2"
                name="line2"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="boite 15"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="PostalAddressCol col-md-4"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="postalCodeGroup"
            >
              <label
                className="form-label"
                htmlFor="postalCode"
              >
                Postal Code
              </label>
              <input
                className="form-control"
                data-testid="postalCode"
                id="postalCode"
                name="postalCode"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="4000"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="PostalAddressCol col-md-8"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="cityGroup"
            >
              <label
                className="form-label"
                htmlFor="city"
              >
                City
              </label>
              <input
                className="form-control"
                data-testid="city"
                id="city"
                name="city"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="Liège"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="PostalAddressCol col-md-12"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="countryGroup"
            >
              <label
                className="form-label"
                htmlFor="country"
              >
                Country
              </label>
              <input
                className="form-control"
                data-testid="country"
                id="country"
                name="country"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="Belgium"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders and compares with same data 1`] = `
<div
  className="AccountInfo"
>
  <div
    className="ComparableField"
  >
    <div
      className="FormGroup"
      style={
        Object {
          "backgroundColor": "#eff3fe",
        }
      }
    >
      <style>
        
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
      </style>
      <div
        data-testid="accountAddressGroup"
      >
        <label
          className="form-label"
          htmlFor="accountAddress"
        >
          some label
        </label>
        <input
          className="form-control"
          data-testid="accountAddress"
          id="accountAddress"
          name="accountAddress"
          onBlur={[Function]}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value="vQxHAE33LeJYV69GCB4o4YcCgnDu8y99u5hy2751fRdxjX9kz"
        />
      </div>
    </div>
  </div>
  <div
    className="Identity"
  >
    <div
      className="row"
    >
      <div
        className="IdentityCol col-md-6"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="firstNameGroup"
            >
              <label
                className="form-label"
                htmlFor="firstName"
              >
                First Name
              </label>
              <input
                className="form-control"
                data-testid="firstName"
                id="firstName"
                name="firstName"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="John"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="IdentityCol col-md-6"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="lastNameGroup"
            >
              <label
                className="form-label"
                htmlFor="lastName"
              >
                Last Name
              </label>
              <input
                className="form-control"
                data-testid="lastName"
                id="lastName"
                name="lastName"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="Doe"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="IdentityCol col-md-6"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="emailGroup"
            >
              <label
                className="form-label"
                htmlFor="email"
              >
                Email
              </label>
              <input
                className="form-control"
                data-testid="email"
                id="email"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="john.doe@logion.network"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="IdentityCol col-md-6"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="phoneNumberGroup"
            >
              <label
                className="form-label"
                htmlFor="phoneNumber"
              >
                Phone Number
              </label>
              <input
                className="form-control"
                data-testid="phoneNumber"
                id="phoneNumber"
                name="phoneNumber"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="+1234"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="PostalAddress"
  >
    <h3>
      Address
    </h3>
    <div
      className="row"
    >
      <div
        className="PostalAddressCol col-md-12"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="line1Group"
            >
              <label
                className="form-label"
                htmlFor="line1"
              >
                Line1
              </label>
              <input
                className="form-control"
                data-testid="line1"
                id="line1"
                name="line1"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="Place de le République Française, 10"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="PostalAddressCol col-md-12"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="line2Group"
            >
              <label
                className="form-label"
                htmlFor="line2"
              >
                Line2
              </label>
              <input
                className="form-control"
                data-testid="line2"
                id="line2"
                name="line2"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="boite 15"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="PostalAddressCol col-md-4"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="postalCodeGroup"
            >
              <label
                className="form-label"
                htmlFor="postalCode"
              >
                Postal Code
              </label>
              <input
                className="form-control"
                data-testid="postalCode"
                id="postalCode"
                name="postalCode"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="4000"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="PostalAddressCol col-md-8"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="cityGroup"
            >
              <label
                className="form-label"
                htmlFor="city"
              >
                City
              </label>
              <input
                className="form-control"
                data-testid="city"
                id="city"
                name="city"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="Liège"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="PostalAddressCol col-md-12"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="countryGroup"
            >
              <label
                className="form-label"
                htmlFor="country"
              >
                Country
              </label>
              <input
                className="form-control"
                data-testid="country"
                id="country"
                name="country"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="Belgium"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders without comparison 1`] = `
<div
  className="AccountInfo"
>
  <div
    className="ComparableField"
  >
    <div
      className="FormGroup"
      style={
        Object {
          "backgroundColor": "#eff3fe",
        }
      }
    >
      <style>
        
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
      </style>
      <div
        data-testid="accountAddressGroup"
      >
        <label
          className="form-label"
          htmlFor="accountAddress"
        >
          some label
        </label>
        <input
          className="form-control"
          data-testid="accountAddress"
          id="accountAddress"
          name="accountAddress"
          onBlur={[Function]}
          onChange={[Function]}
          readOnly={true}
          type="text"
          value="vQxHAE33LeJYV69GCB4o4YcCgnDu8y99u5hy2751fRdxjX9kz"
        />
      </div>
    </div>
  </div>
  <div
    className="Identity"
  >
    <div
      className="row"
    >
      <div
        className="IdentityCol col-md-6"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="firstNameGroup"
            >
              <label
                className="form-label"
                htmlFor="firstName"
              >
                First Name
              </label>
              <input
                className="form-control"
                data-testid="firstName"
                id="firstName"
                name="firstName"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="John"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="IdentityCol col-md-6"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="lastNameGroup"
            >
              <label
                className="form-label"
                htmlFor="lastName"
              >
                Last Name
              </label>
              <input
                className="form-control"
                data-testid="lastName"
                id="lastName"
                name="lastName"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="Doe"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="IdentityCol col-md-6"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="emailGroup"
            >
              <label
                className="form-label"
                htmlFor="email"
              >
                Email
              </label>
              <input
                className="form-control"
                data-testid="email"
                id="email"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="john.doe@logion.network"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="IdentityCol col-md-6"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="phoneNumberGroup"
            >
              <label
                className="form-label"
                htmlFor="phoneNumber"
              >
                Phone Number
              </label>
              <input
                className="form-control"
                data-testid="phoneNumber"
                id="phoneNumber"
                name="phoneNumber"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="+1234"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="PostalAddress"
  >
    <h3>
      Address
    </h3>
    <div
      className="row"
    >
      <div
        className="PostalAddressCol col-md-12"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="line1Group"
            >
              <label
                className="form-label"
                htmlFor="line1"
              >
                Line1
              </label>
              <input
                className="form-control"
                data-testid="line1"
                id="line1"
                name="line1"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="Place de le République Française, 10"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="PostalAddressCol col-md-12"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="line2Group"
            >
              <label
                className="form-label"
                htmlFor="line2"
              >
                Line2
              </label>
              <input
                className="form-control"
                data-testid="line2"
                id="line2"
                name="line2"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="boite 15"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="PostalAddressCol col-md-4"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="postalCodeGroup"
            >
              <label
                className="form-label"
                htmlFor="postalCode"
              >
                Postal Code
              </label>
              <input
                className="form-control"
                data-testid="postalCode"
                id="postalCode"
                name="postalCode"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="4000"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="PostalAddressCol col-md-8"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="cityGroup"
            >
              <label
                className="form-label"
                htmlFor="city"
              >
                City
              </label>
              <input
                className="form-control"
                data-testid="city"
                id="city"
                name="city"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="Liège"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="PostalAddressCol col-md-12"
      >
        <div
          className="ComparableField"
        >
          <div
            className="FormGroup"
            style={
              Object {
                "backgroundColor": "#eff3fe",
              }
            }
          >
            <style>
              
    .FormGroup .form-control,
    .FormGroup .form-control[readonly],
    .FormGroup .input-group .btn-primary {
        color: #000000;
    }
    
            </style>
            <div
              data-testid="countryGroup"
            >
              <label
                className="form-label"
                htmlFor="country"
              >
                Country
              </label>
              <input
                className="form-control"
                data-testid="country"
                id="country"
                name="country"
                onBlur={[Function]}
                onChange={[Function]}
                readOnly={true}
                type="text"
                value="Belgium"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
